packetio/buffer.go

```diff
@@ -10,11 +10,34 @@ import (
 	"github.com/pion/transport/deadline"
 )
 
+// BufferPacketType allow the buffer to know which packet protocol is writing.
+type BufferPacketType int
+
+const (
+	// RTPBufferPacket indicates the buffer that is handling RTP packets
+	RTPBufferPacket BufferPacketType = 1
+	// RTCPBufferPacket indicates the buffer that is handling RTCP packets
+	RTCPBufferPacket BufferPacketType = 2
+)
+
 var errPacketTooBig = errors.New("packet too big")
 
-// Buffer allows writing packets to an intermediate buffer, which can then be read form.
+// Buffer is an interface to implement a writing/reading packet buffer.
+type Buffer interface {
```

Comment:
Nice! Glad to see this was such an easy change. 

Can/should this just embed stdlib interfaces? Read/Write/Close + Deadline I think might already be something.

Reply:
I can set the `io.ReadWriteCloser` but not sure for the other ones